File: androidx/src/androidTest/java/me/panpf/androidx/test/app/DialogxTest.java
package me.panpf.androidx.test.app;

import android.app.Dialog;
import android.os.Bundle;

import androidx.annotation.Nullable;
import androidx.fragment.app.FragmentActivity;
import androidx.test.rule.ActivityTestRule;
import androidx.test.runner.AndroidJUnit4;

import org.junit.Assert;
import org.junit.Rule;
import org.junit.Test;
import org.junit.runner.RunWith;

import me.panpf.androidx.Androidx;
import me.panpf.androidx.app.Dialogx;

@RunWith(AndroidJUnit4.class)
public class DialogxTest {

    @Rule
    public final ActivityTestRule<TestFragmentActivity> mFragmentActivityTestRule =
            new ActivityTestRule<>(TestFragmentActivity.class);

    @Test
    public void testSetClickButtonClosable() {
        Androidx.waitRunInUI(() -> {
            TestFragmentActivity activity = mFragmentActivityTestRule.getActivity();
            Assert.assertTrue(Dialogx.setClickButtonClosable(activity.getDialog(), true));
            Assert.assertTrue(Dialogx.setClickButtonClosable(activity.getDialog(), false));
        });
    }

    @Test
    public void testShowProgressDialog() {
        Androidx.waitRunInUI(() -> {
            TestFragmentActivity activity = mFragmentActivityTestRule.getActivity();
            androidx.fragment.app.Fragment supportFragment = mFragmentActivityTestRule.getActivity()
                    .getSupportFragmentManager()
                    .findFragmentById(android.R.id.content);

            Assert.assertNotNull(Dialogx.showProgressDialog(activity, "by activity"));
            Assert.assertNotNull(Dialogx.showProgressDialog(activity, android.R.string.ok));

            assert supportFragment != null;
            Assert.assertNotNull(Dialogx.showProgressDialog(supportFragment, "by supportFragment"));
            Assert.assertNotNull(Dialogx.showProgressDialog(supportFragment, android.R.string.yes));
        });
    }

    public static class TestFragmentActivity extends FragmentActivity {

        @Override
        protected void onCreate(@Nullable Bundle savedInstanceState) {
            super.onCreate(savedInstanceState);

            getSupportFragmentManager().beginTransaction()
                    .replace(android.R.id.content, new androidx.fragment.app.Fragment())
                    .commit();
        }

        public Dialog getDialog() {
            return new Dialog(this);
        }
    }
}

File: androidx-kt/src/androidTest/java/me/panpf/androidxkt/test/app/DialogxTest.kt
@file:Suppress("DEPRECATION")

package me.panpf.androidxkt.test.app

import android.app.Dialog
import android.os.Bundle
import androidx.test.rule.ActivityTestRule
import androidx.test.runner.AndroidJUnit4
import me.panpf.androidxkt.app.setClickButtonClosable
import me.panpf.androidxkt.app.showProgressDialog
import me.panpf.androidxkt.waitRunInUI
import org.junit.Assert
import org.junit.Rule
import org.junit.Test
import org.junit.runner.RunWith

@RunWith(AndroidJUnit4::class)
class DialogxTest {

    @get:Rule
    val mFragmentActivityTestRule = ActivityTestRule(TestFragmentActivity::class.java)

    @Test
    fun testSetClickButtonClosable() {
        waitRunInUI {
            val activity = mFragmentActivityTestRule.activity
            Assert.assertTrue(activity.dialog.setClickButtonClosable(true))
            Assert.assertTrue(activity.dialog.setClickButtonClosable(false))
        }
    }

    @Test
    fun testShowProgressDialog() {
        waitRunInUI {
            val activity = mFragmentActivityTestRule.activity
            val supportFragment = mFragmentActivityTestRule.activity.supportFragmentManager.findFragmentById(android.R.id.content)

            Assert.assertNotNull(activity.showProgressDialog("by activity"))
            Assert.assertNotNull(activity.showProgressDialog(android.R.string.ok))

            Assert.assertNotNull(supportFragment?.showProgressDialog("by supportFragment"))
            Assert.assertNotNull(supportFragment?.showProgressDialog(android.R.string.yes))
        }
    }

    class TestFragmentActivity : androidx.fragment.app.FragmentActivity() {

        override fun onCreate(savedInstanceState: Bundle?) {
            super.onCreate(savedInstanceState)

            supportFragmentManager.beginTransaction()
                    .replace(android.R.id.content, androidx.fragment.app.Fragment())
                    .commit()
        }

        val dialog: Dialog
            get() = Dialog(this)
    }
}